src/lib/state/activityReducers.ts

```diff
@@ -25,6 +25,35 @@ export const BasicReducers: ActivityReducers = {
     state.list = state.list.filter(f => f.id !== payload.id)
   },
 
+  [ActivityActions.reactivate.type]: (state, { payload }) => {
+    const activity = state.list.find(f => f.id === payload.id);
+    if (activity) {
+      activity.endTime = null;
+    }
+  },
+
+  [ActivityActions.complete.type]: (state, { payload }) => {
+    const activity = state.list.find(f => f.id === payload.id);
+    if (activity) {
+      // First set the end time
+      activity.endTime = payload.endTime;
+
+      // Then clear every participant
+      for (const pId in activity.participants) {
+        const participant = activity.participants[pId]
+        BasicReducers['participant/update'](state, {
+          payload: {
+            activityId: activity.id,
+            participant: { ...participant },
+            update: {
+              time: payload.endTime,
+              status: ResponderStatus.Cleared,
+            }
+          }});

```

Comment:
I was going to say this might be cleaner as something like
```
const action = ActivityActions.participantUpdate(...);
BasicReducers[action.type](state, action);
```
but the activity creator doesn't take a Participant, so you have to blow it all apart. Maybe we want to change `participantUpdate` to take a Participant, and current callers pass firstname/etc as one object?

Reply:
That still feels wrong to me because callers would be forced to update the timeline list themselves before they call the action, so they could have the payload prepped. But that timeline update strongly feels like it should be done in a reducer.

I can't help but wonder if this is a code smell that something is wrong with the data and/or reducer structure. Do we need to flatten the structure a bit?

Proposal: check this in as-is since it does work, but I'll play around with some refactoring options. I can also play around with it more before checking it in.

A couple options I'm thinking of:
- Pull `participants` up to their own top-level slice, alongside `activities`. This has already started to happen with the action names, e.g. `participant/update` rather than `activity/updateParticipant`.
- Keep participants where they are, but pull the timeline up to be a sibling of participants rather than a child.